src/main/java/io/github/novacrypto/base58/EncodeTargetFromCapacity.java

```diff
@@ -0,0 +1,27 @@
+/*
+ *  Base58 library, a Java implementation of Base58 encode/decode
+ *  Copyright (C) 2017-2018 Alan Evans, NovaCrypto
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ *  Original source: https://github.com/NovaCrypto/Base58
+ *  You can contact the authors via github issues.
+ */
+
+package io.github.novacrypto.base58;
+
+public interface EncodeTargetFromCapacity {
```

Comment:
Thought about a `"Factory"` naming here, but nothing requires that it return new instances.